src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -381,8 +395,24 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
             }
 
             bool negExp = false;
-            number.DigitsCount = digEnd;
+            if (digEnd == maxDigCount && numberOfTrailingZeros > 0 && number.HasNonZeroTail)
+            {
+                // We have an additional non-zero digit in the input past maxDigitCount. We already handle this properly.
+                number.DigitsCount = digEnd;
+            }
+            else
+            {
+                // We have only 0s in the fractional part. We can strip it all out
```

Comment:
I think this comment is misleading? It seems to imply this only handles cases like `5.00000` and not cases like `5.1230000`

Reply:
Ah right, good catch. Removing it